pkg/pillar/cmd/loguploader/loguploader.go

```diff
@@ -481,6 +549,7 @@ func doFetchSend(ctx *loguploaderContext, zipDir string, iter *int) int {
 	var isApp bool
 	if zipDir == types.NewlogUploadAppDir {
 		isApp = true
+		appGzipMap = make(map[string]bool)
```

Comment:
Is doFetchSend running in the same goroutine as the checkAppLogMetrics?
If not then you need a lock around the appGzipMap to avoid concurrent read and write by different goroutines,

Reply:
the same go routine.